src/chains.jl

```diff
@@ -748,11 +775,18 @@ function _cat(::Val{3}, c1::Chains, args::Chains...)
     all(c -> names(c) == nms, args) || throw(ArgumentError("chain names differ"))
 
     # concatenate all chains
-    data = mapreduce(c -> c.value.data, (x, y) -> cat(x, y; dims = 3), args;
-                     init = c1.value.data)
+    combined = [c1, args...]
```

Comment:
The current implementation deliberately avoids splatting of `args`. Why is this needed? Could we also check if everything is still inferred correctly?